rtgui/batchqueuepanel.cc

```diff
@@ -322,6 +337,122 @@ void BatchQueuePanel::setGuiFromBatchState(bool queueRunning, int qsize)
+    auto insertTopicHeading = [&pos, buffer](const Glib::ustring& text) {
+        pos = buffer->insert_markup(pos, Glib::ustring::format("\n\n<u><b>", text, "</b></u>\n"));
+    };
+    auto insertTopicBody = [&pos, buffer](const Glib::ustring& text) {
+        pos = buffer->insert_markup(pos, Glib::ustring::format("\n", text, "\n"));
+    };
+    auto mainTitle = M("QUEUE_LOCATION_TEMPLATE_HELP_TITLE");
+    pos = buffer->insert_markup(pos, Glib::ustring::format("<big><b>", mainTitle, "</b></big>\n"));
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_INTRO"));
+
+    insertTopicHeading(M("QUEUE_LOCATION_TEMPLATE_HELP_EXAMPLES_TITLE"));
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_EXAMPLES_BODY"));
+
+    insertTopicHeading(M("QUEUE_LOCATION_TEMPLATE_HELP_PATHS_TITLE"));
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_PATHS_INTRO"));
+    pos = buffer->insert(pos, "\n");
+#ifdef _WIN32
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_PATHS_INTRO_WINDOWS"));
+    pos = buffer->insert(pos, "\n");
+#endif
+    pos = buffer->insert(pos, "\n");
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_PATHS_BODY_1"));
+#ifdef _WIN32
+    auto exampleFilePath = M("QUEUE_LOCATION_TEMPLATE_HELP_PATHS_EXAMPLE_WINDOWS");
+#else
+    auto exampleFilePath = M("QUEUE_LOCATION_TEMPLATE_HELP_PATHS_EXAMPLE_LINUX");
+#endif
+    pos = buffer->insert_markup(pos, Glib::ustring::format("\n  ", exampleFilePath, "\n"));
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_PATHS_BODY_2"));
+    // Examples are generated from exampleFilePath using the actual template processing function
+    Options savedOptions = options; // to be restored after generating example results
+    options.saveUsePathTemplate = true;
+    // Since this code only ever runs once (the first time the help text is presented), no attempt is
+    // made to be efficient. Use a brute-force method to discover the number of elements in exampleFilePath.
+    int pathElementCount = 0;
+    for (int n=9; n>=0; n--) {
+        options.savePathTemplate = Glib::ustring::format("%d",n);
+        auto result = BatchQueue::calcAutoFileNameBase(exampleFilePath);
+        if (!result.empty()) {
+            // The 'd' specifier returns an empty string if N exceeds the number of path elements, so
+            // the largest N that does not return an empty string is the number of elements in exampleFilePath.
+            pathElementCount = n;
+            break;
+        }
+    }
+    // Function inserts examples for a particular specifier, with every valid N value for the
+    // number of elements in the path.
+    auto insertPathExamples = [&buffer, &pos, pathElementCount, exampleFilePath](char letter, int offset1, int mult1, int offset2, int mult2)
+    {
+        for (int n=0; n<pathElementCount; n++) {
+            auto path1 = Glib::ustring::format("%",letter,offset1+n*mult1);
```

Comment:
Space after `,`, please. There are other places in your new code, too, that you should pad a bit more.